proactive-jupyter-kernel/kernel.py

```diff
@@ -1971,9 +1992,96 @@ def __submit_job__(self, input_data):
+        restapi = self.gateway.getProactiveRestApi()
+        hosts = restapi.get_rm_model_hosts()
+
+        html = self.__list_html_table__(hosts)   
+        if html =='':
+            warning_msg = '<div><i class="fa fa-warning"></i> Warning Message: None of node source is available.</div>'
+            if  send_response:
+                self.__display_html__(warning_msg)
+            return  html
+        else:
+            html = '<table border=2 class="stocktable" id="table1" > <th>List of available hots</th>' + html + '</table>'
+
+            if send_response:
+                html = html + '<div class="info-msg"><i class="fa fa-info-circle"></i> If you don\'t set a host name for your job, it will run on any available host.</div>'
+                self.__display_html__(html)
+            else:
+                return html     
+
+    def __get_target_env_tokens__(self, input_data, send_response=True, warning_response=True):
+        restapi = self.gateway.getProactiveRestApi()
+        tokens = restapi.get_rm_model_tokens()
+
+        html = self.__list_html_table__(tokens)   
+        if html =='':
+            warning_msg = '<div><i class="fa fa-warning"></i> Warning Message: None of node source is available.</div>'
+            if  send_response:
+                self.__display_html__(warning_msg)
+            return  html
+        else:
+            html = '<table border=2 class="stocktable" id="table1" > <th>List of available tokens</th>' + html + '</table>'
+           
+            if send_response:
+                html = html + '<div><i class="fa fa-info-circle"></i> If you don\'t set a token name for your job, it will run on any available token.</div>'
+                self.__display_html__(html)
+            else:
+                return html      
+
+    def __get_target_envs__(self, input_data):
+        
+       html_nodesources = self.__get_target_env_nodesources__(input_data, False) 
+       html_hosts = self.__get_target_env_hosts__(input_data, False)
+       html_token = self.__get_target_env_tokens__(input_data, False)
+
+       if html_nodesources !='' or  html_hosts !='' or html_token !='':
+        html = '</div><div id="container"><div id="navigation" style="float:left;padding:10px">' + html_nodesources + '</div><div id="content" style="float:left;padding:10px"><div id="shoutout-box">' + html_hosts  + '</div></div><div id="content" style="float:left;padding:10px"><div id="shoutout-box">' + html_token  + '</div></div></div>'
+        html =  '<style>.footer {clear: both; border-top: 1px solid #000;}</style>' + html + '<div class="footer"><i class="fa fa-info-circle"></i> If you don\'t set a node source, host or token name for your job, it will run on any of them available.</div>'
+        self.__display_html__(html)  
+
+       if html_nodesources == '':
+        warning_msg = '<div><i class="fa fa-warning"></i> Warning Message: None of node source is available.</div>'
+        self.__display_html__(warning_msg)
```

Comment:
I think instead of having separate 3 columns we should link the three information together to know which hosts are related to which NS, and which tokens are related to which host nodes.